### PR TITLE
dmp 1244 removed endpoint

### PR DESCRIPTION
**Pull request by chrisdavecm**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-1244

### Change description ###
Removed delete courthouse endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/darts/courthouses/CourthousesFunctionalTest.java
Comment (chrisbellingham-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/darts/courthouses/CourthousesFunctionalTest.java:175]: State leaks between the Courthouse functional tests, so this is sadly a crucial step because its role is to also clean up the courthouse that was created in a preecding test. So when removing this test, we need to rethink how this test class handles clean up.

I'm somewhat working on this as part of DMP-2279, so might be worth holding this ticket until DMP-2279 is merged in.

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**
